clients/client-java/src/main/java/com/datastrato/gravitino/client/FilesetCatalog.java

```diff
@@ -0,0 +1,253 @@
+   * @param properties The properties of the fileset.
+   * @return The created fileset metadata
+   * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws FilesetAlreadyExistsException If the fileset already exists.
+   */
+  @Override
+  public Fileset createFileset(
+      NameIdentifier ident,
+      String comment,
+      Fileset.Type type,
+      String storageLocation,
+      Map<String, String> properties)
+      throws NoSuchSchemaException, FilesetAlreadyExistsException {
+    NameIdentifier.checkFileset(ident);
+
+    FilesetCreateRequest req =
+        FilesetCreateRequest.builder()
+            .name(RESTUtils.encodeString(ident.name()))
+            .comment(comment)
+            .type(type)
+            .storageLocation(storageLocation)
+            .properties(properties)
+            .build();
+
+    FilesetResponse resp =
+        restClient.post(
+            formatFilesetRequestPath(ident.namespace()),
+            req,
+            FilesetResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.filesetErrorHandler());
+    resp.validate();
+
+    return resp.getFileset();
+  }
+
+  /**
+   * Update a fileset metadata in the catalog.
+   *
+   * @param ident A fileset identifier.
+   * @param changes The changes to apply to the fileset.
+   * @return The updated fileset metadata.
+   * @throws NoSuchFilesetException If the fileset does not exist.
+   * @throws IllegalArgumentException If the changes are invalid.
+   */
+  @Override
+  public Fileset alterFileset(NameIdentifier ident, FilesetChange... changes)
+      throws NoSuchFilesetException, IllegalArgumentException {
+    NameIdentifier.checkFileset(ident);
+
+    List<FilesetUpdateRequest> updates =
+        Arrays.stream(changes)
+            .map(DTOConverters::toFilesetUpdateRequest)
+            .collect(Collectors.toList());
+    FilesetUpdatesRequest req = new FilesetUpdatesRequest(updates);
+
+    FilesetResponse resp =
+        restClient.put(
+            formatFilesetRequestPath(ident.namespace())
+                + "/"
+                + RESTUtils.encodeString(ident.name()),
+            req,
+            FilesetResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.filesetErrorHandler());
+    resp.validate();
+
+    return resp.getFileset();
+  }
+
+  /**
+   * Drop a fileset from the catalog.
+   *
+   * <p>The underlying files will be deleted if this fileset type is managed, otherwise, only the
+   * metadata will be dropped.
+   *
+   * @param ident A fileset identifier.
+   * @return true If the fileset is dropped, false the fileset did not exist.
+   */
+  @Override
+  public boolean dropFileset(NameIdentifier ident) {
+    NameIdentifier.checkFileset(ident);
+
+    DropResponse resp =
+        restClient.delete(
+            formatFilesetRequestPath(ident.namespace())
+                + "/"
+                + RESTUtils.encodeString(ident.name()),
```

Comment:
Why do we encode the `name` for fileset here? I found that we don't do this for metalake, catalog, schema, maybe we should unify them all?

Reply:
Also if we encode the string here, do we need to decode it on the server side?

Reply:
@jerryshao @mchades Actually, I think we can validate the name of a fileset when creating it. Not like a table, the fileset is managed by Gravitino, so we can make sure that the name complies with regulations. But the table, like SAP, we indeed meet some special names that need to be encoded. 

So I think we don't need to encode the name of fileset. Instead, We can verify the legitimacy of the name during creation. What do you think?


The name of SAP table:
![image](https://github.com/datastrato/gravitino/assets/38486782/cd423125-744b-4d77-a2c7-30fdc0d0bb17)

Reply:
> Also if we encode the string here, do we need to decode it on the server side?

As for your question, I think it is not necessary to decode it on the server side. Only if we make sure to decode it on the client, it won't have any impact on users. The only thing I can think of is whether we should decode it before writing it to the audit log.

Reply:
From my point, we don't have to encode the fileset name solely here, we can remove this encoding here, and consider this as a whole solution here https://github.com/datastrato/gravitino/issues/1652